Extensions/TileMap/pixi-tilemap-helper.js

```diff
@@ -0,0 +1,261 @@
+    factory(exports)
+  } else {
+    // Browser globals
+    factory((root.PixiTileMapHelper = {}))
+  }
+})(typeof self !== 'undefined' ? self : this, function (exports) {
+  /**
+   * Tileset/Tilemap related data.
+   */
+  const loadedTileSets = {}
+
+  /**
+   * Creates a Tileset resource from a tiledData json file exported from https://www.mapeditor.org/.
+   * Later on this can potentially be refactored to support other data structures (LED editor for example) https://github.com/deepnight/led
+   */
+  const createTileSetResource = (
+    tiledData,
+    tex,
+    requestedTileSetId,
+    onLoad,
+    getTexture,
+    jsonResourceName
+  ) => {
+    if (!tiledData.tiledversion) {
+      console.warn(
+        "The json data doesn't contain a tiledversion key. Are you sure this file has been exported from mapeditor.org?"
+      )
+      return // TODO handle detecting and loading LDtk tilesets/maps
+    }
+    // This assumes that the tileset is embedded in the tilemap, which it might not always be, so we need to check
+    if (!tiledData.tilesets.length || 'source' in tiledData.tilesets[0]) {
+      console.warn(`
+        ${jsonResourceName} doesn't appear to contain any tileset data.
+      `)
+      return
+    }
+    const {
+      tilewidth,
+      tileheight,
+      tilecount,
+      tiles,
+      image,
+      columns,
+      spacing,
+      margin,
+    } = tiledData.tilesets[0]
+    if (!tex) tex = getTexture(image)
+
+    const rows = tilecount / columns
+    const tileWidth = tilewidth;
+    const tileHeight = tileheight;
+    // We try to detect what size tiled is expecting and if the texture is not the right size, dont load it
+    const expectedAtlasWidth =
+      (tilewidth * columns) + (spacing * (columns - 1)) + (margin * 2)
+    const expectedAtlasHeight =
+      (tileheight * rows) + (spacing * (rows - 1)) + (margin * 2)
+    if (
+      (tex.width !== 1 && expectedAtlasWidth !== tex.width) ||
+      (tex.height !== 1 && expectedAtlasHeight !== tex.height)
+    ) {
+      console.warn(`
+        Have you resized your atlas?
+        It should be ${expectedAtlasWidth}x${expectedAtlasHeight} px, but it's ${tex.width}x${tex.height} px.
+      `)
+      return
+    }
+
+    const textureCache = new Array(tilecount + 1).fill(0).map((_, frame) => {
+      const columnMultiplier = Math.floor((frame - 1) % columns)
+      const rowMultiplier = Math.floor((frame - 1) / columns)
+      const x = (margin + (columnMultiplier * (tileWidth + spacing)))
+      const y = margin + (rowMultiplier * (tileHeight + spacing))
+
+      try {
+        const rect = new PIXI.Rectangle(x, y, tileWidth, tileHeight)
+        const texture = new PIXI.Texture(tex, rect)
+        texture.baseTexture.scaleMode = PIXI.SCALE_MODES.NEAREST
+        texture.cacheAsBitmap = true
+
+        return texture
+      } catch (error) {
+        console.log(rect);
+        console.error(
+          'Error occured while trying to create PIXI.Texture!',
```

Comment:
Give a bit more context, and don't use exclamation marks:
'An error occurred while creating a PIXI.Texture to be used in a TileSet:'